### ISSUE TITLE
Python's string_escape is not captured

### ISSUE THREAD

**Issue by jopasserat** [proot-me/PRoot#71]
This problem was already present in the previous release of CARE and is still there in 2.2.

When trying to archive a python script that somehow, through an import chain, makes use of a package called string_escape, the associated files (string_escape.py and string_escape.pyc) are not captured and archived.

The thing is this package is not explicitly loaded in the code, and it's only referred to as options to function call. For instance, a function in the pickle.py package takes "string-escape" (dash not underscore, for some reason) as a parameter:

```
self.append(rep.decode("string-escape"))
```

Adding the two files to the generated archive solves the problem.

**Comment by jopasserat** [comment]
I managed to make it work with another similar application by having CARE reveal my all home directory.

I guess there is some sort of python configuration in there.

That doesn't explain why it's not detected otherwise though.

**Comment by jopasserat** [comment]
I came across a similar problem with Python3.5 this time.

## TL;DR

I've got an explanation, a workaround, and the TL;DR is: not sure something could be improved in the way CARE works: it's more to do with how Python works so Python users => ~~switch to a better language~~ / be careful :trollface: 

## Full description

Here we go for the whole picture.

A minimal working example to reproduce this would be:

```
# skip output for the first 2 commands
➜ care -r ~ -o /tmp/python3_basic.tgz.bin python3 -c 'print("hello")'
➜ /tmp/python3_basic.tgz.bin

# successful re-execution
➜  /tmp /tmp/python3_basic/re-execute.sh 
hello

# failed re-execution
➜  /tmp /tmp/python3_basic/re-execute.sh > /tmp/foo  
Fatal Python error: Py_Initialize: can't initialize sys standard streams
AttributeError: module '_bootlocale' has no attribute 'getpreferredencoding'

Current thread 0x00007f7c737bb700 (most recent call first):
proot info: pid 10109: terminated with signal 6
```

The problem occurs when Python initialises, it loads a sequence of default packages (you can see them using the `-v` flag on the command line).

As for any other python package, the interpreter (at least CPython) looks first for a previously “compiled” version of the package (the .pyc) and loads it instead of the equivalent .py if it finds any.

_Quoting @ghisvail_

> Whatever module you import (not the executed script),
> the byte code executed by the Python VM will be cached in these .pyc
> files.
> 
> With Python 2, they can be found alongside the corresponding .py files.
> With Python 3, they are stored in a separate __pycache__ folder in the
> folder where the module is located.
> 
> 

The problem of this loading sequence with CARE / Proot is that if the .pyc contains some platform dependent data (it seems to be something with the encoding here), the only the .pyc is captured in the archive. The .py is present but empty, I assume because it wasn’t read.

```
# file's size is 0!!
➜  /tmp find python3_basic/rootfs -name '_bootlocale.py' -exec ls -lh {} \; 
-rw-r--r-- 1 jopasserat users 0 Jan  3 04:40 python3_basic/rootfs/usr/lib/python3.5/_bootlocale.py
```

_Quoting @ghisvail again_

> 
> The cache files contain platform dependent data. I
> believe you can instruct the Python interpreter to not save these cache
> files via an environment variable (PYTHONDONTWRITEBYTECODE), or
> invoking the interpreter with the -B flag.

There's also something I couldn't clearly understand but this problem occurs only when the stdout/err of the Python interpreter are redirected / or captured by another program than the running shell. This is what happens when calling the `re-execute.sh` script from another application or simply by redirecting its output with bash's `>`

## Conclusion 

I'm closing this issue as it doesn't seem related to CARE itself but very specific to some Python intrinsic. More than happy to re-open if someone with more Python-fu knows how to circumvent this from happening when archiving.

Until then, careful pythonists!
